lib/features/login/data/network/config/authorization_interceptors.dart

```diff
@@ -1,32 +1,123 @@
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:core/core.dart';
 import 'package:dio/dio.dart';
 import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/login/data/local/account_cache_manager.dart';
+import 'package:tmail_ui_user/features/login/data/local/token_oidc_cache_manager.dart';
+import 'package:tmail_ui_user/features/login/data/network/oidc_http_client.dart';
 
 class AuthorizationInterceptors extends InterceptorsWrapper {
-  String? _authorization;
+
+  final Dio _dio;
+  final OIDCHttpClient _oidcHttpClient;
+  final TokenOidcCacheManager _tokenOidcCacheManager;
+  final AccountCacheManager _accountCacheManager;
+
+  AuthenticationType _authenticationType = AuthenticationType.none;
+  OIDCConfiguration? _configOIDC;
   Token? _token;
+  String? _authorization;
+
+  AuthorizationInterceptors(
+    this._dio,
+    this._oidcHttpClient,
+    this._tokenOidcCacheManager,
+    this._accountCacheManager
+  );
+
+  void setAuthenticationType(AuthenticationType newAuthenticationType) {
+    _authenticationType = newAuthenticationType;
+  }
+
+  void setOIDCConfiguration(OIDCConfiguration? newConfig) {
+    _configOIDC = newConfig;
+  }
 
   void changeAuthorization(String? userName, String? password) {
     _authorization = base64Encode(utf8.encode('$userName:$password'));
   }
 
   void setToken(Token? newToken) {
     _token = newToken;
+    log('AuthorizationInterceptors::setToken(): newToken: $newToken');
+    log('AuthorizationInterceptors::setToken(): EXPIRE_DATE: ${newToken?.expiredTime?.toIso8601String()}');
   }
 
   @override
   void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
-    if (_authorization != null) {
-      options.headers[HttpHeaders.authorizationHeader] = _getAuthorizationAsBasicHeader(_authorization);
-    } else if (_token != null && _token?.isTokenValid() == true) {
-      options.headers[HttpHeaders.authorizationHeader] = _getTokenAsBearerHeader(_token!.token);
+    switch(_authenticationType) {
+      case AuthenticationType.basic:
+        if (_authorization != null) {
+          options.headers[HttpHeaders.authorizationHeader] = _getAuthorizationAsBasicHeader(_authorization);
+        }
+        break;
+      case AuthenticationType.oidc:
+        if (_token != null && _token?.isTokenValid() == true) {
+          options.headers[HttpHeaders.authorizationHeader] = _getTokenAsBearerHeader(_token!.token);
+        }
+        break;
+      case AuthenticationType.none:
+        break;
     }
     super.onRequest(options, handler);
   }
 
+  @override
+  void onError(DioError err, ErrorInterceptorHandler handler) async {
+    final requestOptions = err.requestOptions;
+    if (_isAuthenticationError(err) && _authenticationType == AuthenticationType.oidc) {
```

Comment:
We will have varied kind of authentication Error, please more specific in the method name. `_isTokenExpired`

Reply:
not get your point.

Reply:
````dart
if ((_isTokenExpired() || err.response?.statusCode == 401) 
   && _authenticationType == AuthenticationType.oidc) {
 
}
````